### PR TITLE
repos: add Pagure code host support

### PR DESCRIPTION
**Pull request by tsenart**
This commit adds Pagure code host support, disabled by default in site
config. This will allow us to sync repos from src.fedoraproject.org on
sourcegraph.com.

Part of #28028

### CODE REVIEW

File: internal/repos/pagure.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
+package repos
+
+import (
+	"context"
+	"path"
+	"strconv"
+
+	"github.com/cockroachdb/errors"
+	"github.com/goware/urlx"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/pagure"
```
Comment (ryanslade) [inline internal/repos/pagure.go:10]: nit: Should be grouped below

File: internal/repos/pagure.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		Tags:      s.cli.Config.Tags,
+		Pattern:   s.cli.Config.Pattern,
+		Namespace: s.cli.Config.Namespace,
+		Fork:      s.cli.Config.Forks,
+	}
+
+	for {
+		page, err := s.cli.ListProjects(ctx, args)
+		if err != nil {
+			results <- SourceResult{Source: s, Err: err}
```
Comment (ryanslade) [inline internal/repos/pagure.go:69]: Do we need to bail out if the context is cancelled while sending on the channel here or lower down?

File: internal/repos/pagure_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
+package repos
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
```
Comment (ryanslade) [inline internal/repos/pagure_test.go:10]: nit: Import grouping

**Review by ryanslade — APPROVED** [review]

File: internal/repos/pagure.go (hunk 2 above)
Comment (tsenart) [reply internal/repos/pagure.go:69]: I don't think so: https://github.com/sourcegraph/sourcegraph/blob/c0a129d0ba096c1497ae9c9cf1bf37e7ddece0b7/internal/repos/syncer.go#L421-L424

Looks good to you too?

Comment (ryanslade) [reply internal/repos/pagure.go:69]: Yeah, it looks like it should be ok but makes an assumption that both the sender and receiver will stop work on the first error.

Comment (tsenart) [reply internal/repos/pagure.go:69]: > but makes an assumption that both the sender and receiver will stop work on the first error.

Can you help me understand how that assumption is encoded in the code?

**Pull request merged**